tests/extension/oneapi_root_group/root_group.cpp

```diff
@@ -0,0 +1,256 @@
+
+  // get_local_linear_range
+  CHECK(std::is_same_v<decltype(std::declval<rg>().get_local_linear_range()),
+                       size_t>);
+  CHECK(results[checks::get_local_linear_range]);
+
+  // leader
+  CHECK(std::is_same_v<decltype(std::declval<rg>().leader()), bool>);
+  CHECK(results[checks::leader]);
+
+  using sycl::ext::oneapi::experimental::get_child_group;
+
+  // get_child_group(root_group)
+  CHECK(std::is_same_v<decltype(get_child_group(std::declval<rg>())),
+                       sycl::group<Dimensions>>);
+  CHECK(results[checks::get_child_group___root_group]);
+
+  // get_child_group(group)
+  CHECK(std::is_same_v<decltype(get_child_group(
+                           std::declval<sycl::group<Dimensions>>())),
+                       sycl::sub_group>);
+  CHECK(results[checks::get_child_group___group]);
+}
+
+template <typename KernelName, size_t... Dims>
+static void check_root_group_barrier() {
+  constexpr int Dimensions = sizeof...(Dims);
+  auto q = sycl_cts::util::get_cts_object::queue();
+  auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  auto kernel = bundle.get_kernel<KernelName>();
+  auto maxWGs = kernel.template ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  REQUIRE(maxWGs >= 1);
+  auto local_range = sycl::range<Dimensions>(Dims...);
+  auto global_range = local_range;
+  global_range[0] *= maxWGs;
+  REQUIRE(global_range.size() == local_range.size() * maxWGs);
+  auto nd_range = sycl::nd_range<Dimensions>{global_range, local_range};
+  auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+
+  const int N = static_cast<int>(global_range.size());
+  int* data = sycl::malloc_shared<int>(N, q);
+  std::fill_n(data, N, 0);
+
+  q.parallel_for<KernelName>(nd_range, props, [=](auto it) {
+    auto root = it.ext_oneapi_get_root_group();
+    data[root.get_local_linear_id()] = 1;
+    sycl::group_barrier(root);
+    int sum = 0;
+    for (int i = N - 1; i >= 0; i--) {
+      sum += data[i];
+    }
+    sycl::group_barrier(root);
+    data[root.get_local_linear_id()] = (sum == N ? 1 : 0);
+  });
+  q.wait();
+
+  for (int i = 0; i < N; i++) {
+    CHECK(data[i] == 1);
+  }
+  sycl::free(data, q);
+}
+
+#endif
+
+TEST_CASE(
+    "Test for \"Root Group\" extension, check root_group class and "
+    "get_child_group API functionality",
+    "[oneapi_root_group_api]") {
+#ifndef SYCL_EXT_ONEAPI_ROOT_GROUP
+  SKIP("SYCL_EXT_ONEAPI_ROOT_GROUP is not defined");
+#else
+  check_root_group_api<struct RootGroup1D, 4>();
+  check_root_group_api<struct RootGroup2D, 6, 4>();
+  check_root_group_api<struct RootGroup3D, 8, 6, 4>();
+#endif
+}
+
+TEST_CASE(
+    "Test for \"Root Group\" extension, check root_group synchronization using "
+    "group_barrier function"
+    "[oneapi_root_group_barrier]") {
```

Comment:
I wonder if those two test cases should instead have the same tag which corresponds to the extension, to be able to specifically launch just those two test cases?

Whether we need extra tags for additional filtering (like `api` or `barrier`) I'm not sure - this test looks small enough so that doesn't seem necessary. No idea if there is some CTS-wide guideline about tags usage, any recommendations from other reviewers?

Reply:
> I didn't know it's valid to reuse a tag

Based on the [documentation](https://github.com/catchorg/Catch2/blob/v2.x/docs/test-cases-and-sections.md#tags), those tags don't have to be unique, but rather intended to provide functionality to group tests into various categories to have more flexibility when launching them.